### PR TITLE
Retain series name in `Series.apply`

### PR DESCRIPTION
**Pull request by brandon-b-miller**
Closes https://github.com/rapidsai/cudf/issues/10715

### CODE REVIEW

**Review by shwina — APPROVED** [review]

**Comment by shwina** [comment]
@gpucibot merge

File: python/cudf/cudf/core/series.py
Comment (mmccarty) [inline python/cudf/cudf/core/series.py:2114]: Perhaps this can be done in `_apply`?

Comment (brandon-b-miller) [reply python/cudf/cudf/core/series.py:2114]: I had considered this initially the problem being that `DataFrame` also inherits `_apply` from `IndexedFrame`, and `DataFrame` objects have no `name`.

**Comment by brandon-b-miller** [comment]
rerun tests

File: python/cudf/cudf/tests/test_udf_masked_ops.py
Comment (bdice) [inline python/cudf/cudf/tests/test_udf_masked_ops.py:374]: Ideally we should avoid GPU allocations in the parametrize calls.

```python
@pytest.mark.parametrize(
    "data,name",
    [
        ([1, 2, 3], None),
        ([1, cudf.NA, 3], None),
        ([1, 2, 3], "test_name"),
    ],
)
def test_series_apply_basic(data, name):
    s = cudf.Series(data, name=name)

    def func(x):
        return x + 1

    run_masked_udf_series(func, s, check_dtype=False)
```

File: python/cudf/cudf/core/series.py
Comment (mmccarty) [reply python/cudf/cudf/core/series.py:2114]: True, but you could add a `hasattr` check.

**Comment by brandon-b-miller** [comment]
@bdice @mmccarty could one of you request changes to block this from merging since gpucibot got the magic words already?

**Review by mmccarty — CHANGES_REQUESTED** [review]
Please encapsulate in `_apply` and add `hasattr` check for `name` with comment explaining that `dataframe` has no `name` attribute.

**Review by bdice — CHANGES_REQUESTED** [review]
(I don't think Mike is a code owner so his change request may not block merger)
image

File: python/cudf/cudf/core/series.py
Comment (bdice) [reply python/cudf/cudf/core/series.py:2114]: I agree this is best handled in a post-processing step specific to `Series` rather than trying to check for attributes in a parent class that is not an effective model of its children. The current change in this PR (`result.name = self.name`) is fine with me.

Comment (shwina) [reply python/cudf/cudf/core/series.py:2114]: We don't use `_apply()` internally, but if we did, I can see the benefit of correct name handling. I don't feel strongly here. Defer to Mike on the final word

**Comment by mmccarty** [comment]
Figured it was worth a try! :D

File: python/cudf/cudf/core/series.py
Comment (mmccarty) [reply python/cudf/cudf/core/series.py:2114]: :100: agree with using inheritance here. Thanks @shwina!

**Comment by mmccarty** [comment]
> Please encapsulate in `_apply` and add `hasattr` check for `name` with comment explaining that dataframe has no name attribute.

Or just use inheritance as @shwina suggests.

**Review by bdice — APPROVED** [review]
Thanks @brandon-b-miller!

**Comment by brandon-b-miller** [comment]
rerun tests

**Pull request merged**
